rpcs3/rpcs3qt/user_manager_dialog.cpp

```diff
@@ -0,0 +1,359 @@
+
+	//m_table->setItemDelegate(new table_item_delegate(this)); // to get rid of cell selection rectangles include "table_item_delegate.h"
+	m_table->setSelectionMode(QAbstractItemView::SelectionMode::SingleSelection);
+	m_table->setSelectionBehavior(QAbstractItemView::SelectRows);
+	m_table->setContextMenuPolicy(Qt::CustomContextMenu);
+	m_table->setColumnCount(2);
+	m_table->setHorizontalHeaderLabels(QStringList() << tr("User ID") << tr("User Name"));
+
+	QPushButton* push_remove_user = new QPushButton(tr("Delete User"), this);
+	QPushButton* push_create_user = new QPushButton(tr("Create User"), this);
+	QPushButton* push_login_user = new QPushButton(tr("Log In User"), this);
+	QPushButton* push_rename_user = new QPushButton(tr("Rename User"), this);
+
+	QPushButton* push_close = new QPushButton(tr("&Close"), this);
+	push_close->setAutoDefault(true);
+
+	// Button Layout
+	QHBoxLayout* hbox_buttons = new QHBoxLayout();
+	hbox_buttons->addWidget(push_create_user);
+	hbox_buttons->addWidget(push_login_user);
+	hbox_buttons->addWidget(push_rename_user);
+	hbox_buttons->addWidget(push_remove_user);
+	hbox_buttons->addStretch();
+	hbox_buttons->addWidget(push_close);
+
+	// main layout
+	QVBoxLayout* vbox_main = new QVBoxLayout();
+	vbox_main->setAlignment(Qt::AlignCenter);
+	vbox_main->addWidget(m_table);
+	vbox_main->addLayout(hbox_buttons);
+	setLayout(vbox_main);
+
+	user_manager_dialog::UpdateTable();
+
+	QByteArray geometry = m_gui_settings->GetValue(gui::um_geometry).toByteArray();
+	if (geometry.isEmpty() == false)
+	{
+		restoreGeometry(geometry);
+	}
+
+	// Connects and events
+	connect(push_close, &QAbstractButton::clicked, this, &user_manager_dialog::close);
+	connect(push_remove_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserRemove);
+	connect(push_create_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserCreate);
+	connect(push_rename_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserRename);
+	connect(push_login_user, &QAbstractButton::clicked, this, &user_manager_dialog::OnUserLogin);
+	connect(m_table, &QTableWidget::itemDoubleClicked, this, &user_manager_dialog::OnUserLogin);
+	connect(m_table->horizontalHeader(), &QHeaderView::sectionClicked, this, &user_manager_dialog::OnSort);
+	connect(m_table, &QTableWidget::customContextMenuRequested, this, &user_manager_dialog::ShowContextMenu);
+	connect(m_table, &QTableWidget::itemSelectionChanged, [=]
+	{
+		int idx = m_table->currentRow();
+		if (idx < 0)
+		{
+			return;
+		}
+		int idx_real = m_table->item(idx, 0)->data(Qt::UserRole).toInt();
+
+		bool enableButtons = m_user_list[idx_real]->GetUserId() != Emu.GetUsr();
+		push_rename_user->setEnabled(enableButtons);
+		push_remove_user->setEnabled(enableButtons);
+		push_login_user->setEnabled(enableButtons);
+	});
+}
+
+void user_manager_dialog::UpdateTable()
+{
+	if (m_dir == "")
+	{
+		// fmt::format(%shome ... is harder to read than straight concatenation.
```

Comment:
lol. funny remark. but you can just use concat. there's nothing wrong with it in this case.
same goes for the changes you made in the last commit. but that's really up to you